src/test/java/com/vinylteam/vinyl/service/impl/DefaultOfferServiceTest.java

```diff
@@ -158,4 +161,49 @@ void getListOfShopsIdsWithNullOffersListTest() {
         assertTrue(actualIds.isEmpty());
     }
 
+    @Test
+    @DisplayName("Checks that offers are merged during dynamic update")
+    void mergeOfferChangesTest() {
+        //when
+        VinylParser parser = new CloneNlParser();
+        Offer oldOffer = dataGenerator.getOfferWithUniqueVinylIdAndShopId(12, 4);
+        RawOffer newlyLoadedOffer = dataGenerator.getRawOfferWithShopIdAndNumber(4, 12);
+        var newPrice = newlyLoadedOffer.getPrice() - 2d;
+        newlyLoadedOffer.setPrice(newPrice);
+        assertNotEquals(newlyLoadedOffer.getPrice(), oldOffer.getPrice());
+        offerService.mergeOfferChanges(oldOffer, parser, newlyLoadedOffer);
+        //then
+        assertEquals(newlyLoadedOffer.getPrice(), oldOffer.getPrice());
+        assertEquals(newPrice, oldOffer.getPrice());
+    }
+
+    @Test
+    @DisplayName("Checks that offers are merged correctly with non valid offer (as though the link is no more existing)")
+    void mergeOfferChangesWithNonValidOfferTest() {
+        //when
+        VinylParser parser = new CloneNlParser();
+        Offer oldOffer = dataGenerator.getOfferWithUniqueVinylIdAndShopId(12, 4);
+        RawOffer newlyLoadedOffer = new RawOffer();
+        assertTrue(oldOffer.isInStock());
+
+        offerService.mergeOfferChanges(oldOffer, parser, newlyLoadedOffer);
+        //then
+        assertFalse(oldOffer.isInStock());
```

Comment:
Same as before - actualOffer.
Valididty isn't about inStock, it's about is offer empty||filled with default values.

Reply:
if dynamically received offer is not valid, the  mergeOfferChanges() method will just set inStock flag to false, that is all